versioned_docs/version-latest/learn/monitoring.md

```diff
@@ -0,0 +1,723 @@
+:::note
+Only for node running in Docker!
+:::
+
+
+The Subspace Node provides it's metrics on port 9615 via prometheus client libraries.
+The following `yaml` file is designed to scrape Subspace Node metrics when the Node is run in Docker.
+
+We add following lines to the end of node service section in `Docker-Compose.yml` file:
+```bash
+"--prometheus-external"
+"--prometheus-port", "9615"
+```
+and
+```bash
+expose:
+- "9615:9615"
+```
+
+### The `Docker-Compose.yml` for Monitoring the Subspace Node
+
+:::note
+If you're using an external Prometheus located on a different server, the setup might exclude the monitoring stack configuration provided in this template.
+:::
+
+> The following section should be copied from the [official Subspace Docker Compose
+guide](https://docs.subspace.network/docs/farming-&-staking/farming/advanced-cli/cli-install)
+
+```bash
+version: "3.7"
+services:
+  node:
+    image: ghcr.io/subspace/node:gemini-3g-2023-dec-22
+    volumes:
+      - node-data:/var/subspace:rw
+    ports:
+      - "0.0.0.0:30333:30333/udp"
+      - "0.0.0.0:30333:30333/tcp"
+      - "0.0.0.0:30433:30433/udp"
+      - "0.0.0.0:30433:30433/tcp"
+    restart: unless-stopped
+    command:
+      [
+        "--chain", "gemini-3g",
+        "--base-path", "/var/subspace",
+        "--blocks-pruning", "256",
+        "--state-pruning", "archive-canonical",
+        "--port", "30333",
+        "--dsn-listen-on", "/ip4/0.0.0.0/udp/30433/quic-v1",
+        "--dsn-listen-on", "/ip4/0.0.0.0/tcp/30433",
+        "--rpc-cors", "all",
+        "--rpc-methods", "unsafe",
+        "--rpc-external",
+        "--no-private-ipv4",
+        "--validator",
+        "--name", "subspace"
+#Prometheus port:
+        "--prometheus-external"
+        "--prometheus-port", "9615",
+
+      ]
+    healthcheck:
+      timeout: 5s
+      interval: 30s
+      retries: 60
+#This maps port 9615 on the host where Prometheus is running to the same port 9615 within Docker container.
+    expose:
+      - "9615:9615"
+  
+  farmer:
+    depends_on:
+      node:
+        condition: service_healthy
+    image: ghcr.io/subspace/farmer:gemini-3g-2023-dec-22
+    volumes:
+      - farmer-data:/var/subspace:rw
+    ports:
+      - "0.0.0.0:30533:30533/udp"
+      - "0.0.0.0:30533:30533/tcp"
+    restart: unless-stopped
+    command:
+      [
+        "farm",
+        "--node-rpc-url", "ws://node:9944",
+        "--listen-on", "/ip4/0.0.0.0/udp/30533/quic-v1",
+        "--listen-on", "/ip4/0.0.0.0/tcp/30533",
+        "--reward-address", "st7GBwATPqtBkK5y4uXbV52euszPpFPw7wmkF8FywEqJaf8uP",
+        "path=/var/subspace,size=100G"
+      ]
+volumes:
+  node-data:
+  farmer-data:            
+```
+> The above section should be copied from the [official Subspace Docker Compose
+guide](https://docs.subspace.network/docs/farming-&-staking/farming/advanced-cli/cli-install)
+    
+> The below section defines the containers we need for the monitoring stack
```

Comment:
```suggestion
The below section defines the containers we need for the monitoring stack
```

Reply:
I think we should somehow highlight these tips that these are two parts of one big file, and that they need to be compiled separately

Reply:
So I try to use up and down arrows

Reply:
Hard to see how this renders without a fresh preview.